text/0793-polymporphic-relations-without-inheritance.md

```diff
@@ -0,0 +1,135 @@
+---
+Stage: Accepted
+Start Date: 
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): 
+RFC PR: https://github.com/emberjs/rfcs/pull/793
+---
+
+<!--- 
+Directions for above: 
+
+Stage: Leave as is
+Start Date: 2022-02-11
+Release Date: Leave as is
+Release Versions: Leave as is
+Relevant Team(s): Ember Data
+RFC PR: https://github.com/emberjs/rfcs/pull/793
+-->
+
+# Polymorphic relations without inheritance
+
+## Summary
+
+Allow setting polymorph relations without inheritance and mixins.
```

Comment:
We should also explicitly deprecate Mixin and Inheritance based polymorphism as part of this PR. That deprecation would target 5.0 and would only be printed if the relationship definition was not sufficient for determining polymorphism without making use of checks of the prototype chain. E.g. users can continue to use mixins and inheritance to sprinkle in polymorphism if they so desire; however, the mechanism will be simply the actual definition of the relationship and anywhere that fails to be sufficient will print a deprecation asking the user to update the relationship definition to one that is sufficient.